packages/babel-types/src/definitions/es2015.js

```diff
@@ -399,19 +399,27 @@ export const classMethodOrPropertyCommon = {
     optional: true,
   },
   key: {
-    validate: (function() {
-      const normal = assertNodeType(
+    validate: chain(
```

Comment:
Why are these changes needed?

Reply:
It's for the validator parser to be able to guess the apparent type of the `key` key. As it is right now, it's just an opaque arbitrary function; using `chain` and `assertNodeType` lets the type generator generate `Identifier | StringLiteral | NumericLiteral | Expression`.